Preparing vSphere for EKS-A (#1354)
aws/eks-anywhere: docs/content/en/docs/reference/vsphere/vsphere-preparation.md

---
title: "Preparing vSphere for EKS Anywhere"
weight: 20
description: >

>   Set up a VSphere cluster to prepare it for EKS Anywhere

Reviewer: ```suggestion
  Set up a vSphere cluster to prepare it for EKS Anywhere
```

---

## Create a VM and template folder (Optional):
For each user that needs to create workload clusters, have the vSphere administrator create a VM and template folder.
That folder will host:

> * The VMs of the Controlplane and Data plane nodes of each cluster.

Reviewer: ```suggestion
* The VMs of the Control plane and Data plane nodes of each cluster.
```

* A nested folder for the management cluster and another one for each workload cluster.
* Each cluster VM in its own nested folder under this folder.

User permissions should be set up to: 

* Only allow the user to see and create EKS Anywhere resources in that folder and its nested folders.

> * Prevent the user from having visibility and control over the whole vSphere cluster domain and its sub-child objects (data center, resource pools and other folders).

Reviewer: ```suggestion
* Prevent the user from having visibility and control over the whole vSphere cluster domain and its sub-child objects (datacenter, resource pools and other folders).
```

In your EKS Anywhere configuration file you will reference to a path under this folder associated with the cluster you create.

### Add a vSphere folder
Follow these steps to create the user's vSphere folder:

1. In the vSphere Client, select the VM and Template tab.
1. Select either a data center or another folder as a parent object for the folder that you want to create.
1. Right-click the parent object and click New Folder.
1. Enter a name for the folder and click OK.
   For more details, see the [vSphere Create a Folder](https://docs.vmware.com/en/VMware-vSphere/7.0/com.vmware.vsphere.vcenterhost.doc/GUID-031BDB12-D3B2-4E2D-80E6-604F304B4D0C.html) documentation.

### Set up vSphere roles and user permission

> You need to get a VSphere username with the right privileges to let you creatie EKS Anywhere clusters on top of your VSphere cluster.

Reviewer: ```suggestion
You need to get a vSphere username with the right privileges to let you creatie EKS Anywhere clusters on top of your vSphere cluster.
```

Reviewer 2: .

   The default deployment location for the virtual machine is the inventory object where you started the wizard, which is the folder you created above. Click Next.
1. On the Select a compute resource page, select the resource pool where to run the deployed VM template, and click Next. 
1. On the Review details page, verify the OVF or OVA template details and click Next.
1. On the Select storage page, select a datastore to store the deployed OVF or OVA template and click Next.
1. On the Select networks page, select a source network and map it to a destination network. Click Next.
1. On the Ready to complete page, review the page and click Finish.
   For details, see [Deploy an OVF or OVA Template](https://docs.vmware.com/en/VMware-vSphere/7.0/com.vmware.vsphere.vm_admin.doc/GUID-17BEDA21-43F6-41F4-8FB2-E01D275FE9B4.html)

To build your own Ubuntu OVA template check the Building your own Ubuntu OVA section in the following [link]({{< relref "../artifacts/" >}}).

> ## Tagging the Deployed OVA Template

Reviewer 2: > 
Yes, but in my case it was failing downloading it during the creation of the cluster, and I had to download it and import it to Vsphere manually.
